modules/tinymce/src/themes/silver/main/ts/ui/context/ContextForm.ts

```diff
@@ -48,9 +49,18 @@ const buildInitGroup = <T>(
 
 const buildInitGroups = (ctx: InlineContent.ContextForm, providers: UiFactoryBackstageProviders): ToolbarGroup[] => {
   switch (ctx.type) {
-    case 'contextform': return buildInitGroup(Fun.curry(ContextFormTextInput.renderContextFormTextInput, ctx), ctx, providers);
-    case 'contextsliderform': return buildInitGroup(Fun.curry(ContextFormSlider.renderContextFormSliderInput, ctx), ctx, providers);
-    case 'contextsizeinputform': return buildInitGroup(Fun.curry(ContextFormSizeInput.renderContextFormSizeInput, ctx), ctx, providers);
+    case 'contextform': {
+      const valueState = Singleton.value<string>();
```

Comment:
Since all these code paths all execute buildINitGroup and we don't need the singleton for anything else we can move the creation of that into the `buildInitGroup`.